File: patterns/java/src/refactoring_guru/patterns/builder/example/Director.java
package refactoring.guru.patterns.builder.example;

import refactoring.guru.patterns.builder.example.builders.Builder;
import refactoring.guru.patterns.builder.example.cars.Type;
import refactoring.guru.patterns.builder.example.components.Engine;
import refactoring.guru.patterns.builder.example.components.GPSNavigator;
import refactoring.guru.patterns.builder.example.components.Transmission;
import refactoring.guru.patterns.builder.example.components.TripComputer;

public class Director {

    public void constructSportCar(Builder builder) {
        builder.setType(Type.SPORT_CAR);
        builder.setSeats(2);
        builder.setEngine(new Engine(3.0, 0));
        builder.setTransmission(Transmission.SEMI_AUTOMATIC);
        builder.setTripComputer(new TripComputer());
        builder.setGPSNavigator(new GPSNavigator());
    }

    public void constructCityCar(Builder builder) {
        builder.setType(Type.CITY_CAR);
        builder.setSeats(5);
        builder.setEngine(new Engine(1.8, 0));
        builder.setTransmission(Transmission.MANUAL);
        builder.setTripComputer(new TripComputer());
        builder.setGPSNavigator(new GPSNavigator());
    }

    public void constructCabrioletCar(Builder builder) {
        builder.setType(Type.CABRIOLET);
        builder.setSeats(4);
        builder.setEngine(new Engine(2.5, 0));
        builder.setTransmission(Transmission.AUTOMATIC);
        builder.setTripComputer(new TripComputer());
        builder.setGPSNavigator(new GPSNavigator());
    }
}

File: patterns/cs/src/RefactoringGuru/Builder/Example/Director/Director.cs
using Builder.Example.Components;

namespace Builder.Example.Director
{
    /// EN: Director defines the order of building steps. It works with a builder
	/// object through common Builder interface. Therefore it may not know what
	/// product is being built.
	/// 
	/// RU: Директор знает в какой последовательности заставлять работать строителя.
	/// Он работает с ним через общий интерфейс Строителя. Из-за этого, он может не
	/// знать какой конкретно продукт сейчас строится.
	public class Director
    {
        public void ConstructSportsCar(Builders.Builder builder)
        {
            builder.SetType(Cars.Type.SPORTS_CAR);
            builder.SetSeats(2);
            builder.SetEngine(new Engine(3.0, 0));
            builder.SetTransmission(Transmission.SEMI_AUTOMATIC);
            builder.SetTripComputer(new TripComputer());
            builder.SetGpsNavigator(new GPSNavigator());
        }

        public void ConstructCityCar(Builders.Builder builder)
        {
            builder.SetType(Cars.Type.CITY_CAR);
            builder.SetSeats(2);
            builder.SetEngine(new Engine(1.2, 0));
            builder.SetTransmission(Transmission.AUTOMATIC);
            builder.SetTripComputer(new TripComputer());
            builder.SetGpsNavigator(new GPSNavigator());
        }

        public virtual void ConstructSuv(Builders.Builder builder)
        {
            builder.SetType(Cars.Type.SUV);
            builder.SetSeats(4);
            builder.SetEngine(new Engine(2.5, 0));
            builder.SetTransmission(Transmission.MANUAL);
            builder.SetGpsNavigator(new GPSNavigator());
        }
    }
}
